Refactor(eos_designs)!: loopback_ip_pools will take precedence over loopback_ip_range or loopback_ipv6_range (#6317)
aristanetworks/ansible-avd: docs/porting-guides/6.x.x.md

+ evpn_prevent_readvertise_to_server_mode: source_peer_asn
```

### Change of default behavior for `loopback_ip_pools`, `loopback_ip_range` and `loopback_ipv6_range`

> In AVD 5.x.x, `loopback_ip_range` for IPv4 and `loopback_ipv6_range` for IPv6 takes precedence over `loopback_ip_pools`.

Reviewer: ```suggestion
In AVD 5.x.x, `loopback_ip_range` for IPv4 and `loopback_ipv6_range` for IPv6 take precedence over `loopback_ip_pools`.
```

+ evpn_prevent_readvertise_to_server_mode: source_peer_asn
```

> ### Change of default behavior for `loopback_ip_pools`, `loopback_ip_range` and `loopback_ipv6_range`

Reviewer 2: ```suggestion
### `vtep_diagnostic` change of default behavior for `loopback_ip_pools`, `loopback_ip_range` and `loopback_ipv6_range` under network services.
```

> In AVD 5.x.x, `loopback_ip_range` for IPv4 and `loopback_ipv6_range` for IPv6 take precedence over `loopback_ip_pools`.

Reviewer 2: ```suggestion
In AVD 5.x.x, `loopback_ip_range` for IPv4 and `loopback_ipv6_range` for IPv6 take precedence over `loopback_ip_pools` under `vtep_diagnostic` under network services.
```

> As of AVD 6.0.0, if `pod_name` is defined then `loopback_ip_pools` will take precedence over `loopback_ip_range` for IPv4 and `loopback_ipv6_range` for IPV6.

Reviewer 2: ```suggestion
As of AVD 6.0.0, if the POD name defined for a device is matching an entry under `loopback_ip_pools`, this entry will take precedence over `loopback_ip_range` for IPv4 and `loopback_ipv6_range` for IPV6.
```

> To retain the previous behavior do not define either `pod_name` or `loopback_ip_pools`.

Reviewer 2: ```suggestion
To retain the previous behavior remove the pool under `loopback_ip_pools` where `pod` is matching `pod_name`.
```

Author: updated

```diff
+ pod_name: TENANT_NETWORKS_POD
  tenants:
    - name: Tenant_X
      mac_vrf_vni_base: 11000
      vrfs:
        - name: VRF_with_loopback_ip_pools
          vrf_id: 22
          vtep_diagnostic:
            loopback: 200
+           loopback_ip_pools:
+             - pod: TENANT_NETWORKS_POD
+               ipv4_pool: 10.101.102.0/24
+               ipv6_pool: 2001:db8:1::-2001:db8:1:7fff::, 2001:db8:1:8000::/49
+           # loopback_ip_pools will take precedence over loopback_ip_range.

>             loopback_ip_range: 10.255.1.0/24

Reviewer 2: this looks wrong. You say do not define and your are adding stuff with `+`.
Also removing the pod name may be a bad move. Need to think a bit more about this

Reviewer 2: ```suggestion
  pod_name: TENANT_NETWORKS_POD
  tenants:
    - name: Tenant_X
      mac_vrf_vni_base: 11000
      vrfs:
        - name: VRF_with_loopback_ip_pools
          vrf_id: 22
          vtep_diagnostic:
            loopback: 200
            loopback_ip_pools:
-             - pod: TENANT_NETWORKS_POD
-               ipv4_pool: 10.101.102.0/24
-               ipv6_pool: 2001:db8:1::-2001:db8:1:7fff::,2001:db8:1:8000::/49
              - pod: OTHER_POD
                ipv4_pool: 10.101.103.0/24
            loopback_ip_range: 10.255.1.0/24
```

+ evpn_prevent_readvertise_to_server_mode: source_peer_asn
```

### `vtep_diagnostic` change of default behavior for `loopback_ip_pools`, `loopback_ip_range` and `loopback_ipv6_range` under network services

In AVD 5.x.x, `loopback_ip_range` for IPv4 and `loopback_ipv6_range` for IPv6 take precedence over `loopback_ip_pools` under `vtep_diagnostic` under network services.

As of AVD 6.0.0, if the POD name defined for a device is matching an entry under `loopback_ip_pools`, this entry will take precedence over `loopback_ip_range` for IPv4 and `loopback_ipv6_range` for IPV6.

To retain the previous behavior remove the pool under `loopback_ip_pools` where `pod` is matching `pod_name`.

```diff
  pod_name: TENANT_NETWORKS_POD
  tenants:
    - name: Tenant_X
      mac_vrf_vni_base: 11000
      vrfs:
        - name: VRF_with_loopback_ip_pools
          vrf_id: 22
          vtep_diagnostic:
            loopback: 200
            loopback_ip_pools:
-             - pod: TENANT_NETWORKS_POD
-               ipv4_pool: 10.101.102.0/24

> -               ipv6_pool: 2001:db8:1::-2001:db8:1:7fff::,2001:db8:1:8000::/49

Reviewer 3: I think this line should be removed as this will remove the ipv6 pool from configuration which is in contrast to the statement `retain previous behavior`.

Author: Updated

> -               ipv6_pool: 2001:db8:1::-2001:db8:1:7fff::,2001:db8:1:8000::/49

Reviewer 3: ```suggestion
```

Author: Updated
